cmake/kokkos_arch.cmake

```diff
@@ -1139,3 +1139,14 @@ MESSAGE(STATUS "Architectures:")
 FOREACH(Arch ${KOKKOS_ENABLED_ARCH_LIST})
   MESSAGE(STATUS " ${Arch}")
 ENDFOREACH()
+
+
+IF(KOKKOS_ENABLE_ATOMICS_BYPASS)
+  IF(NOT _HOST_PARALLEL STREQUAL "NoTypeDefined" OR NOT _DEVICE_PARALLEL STREQUAL "NoTypeDefined")
+    MESSAGE(FATAL_ERROR "Not allowed to disable atomics if any host parallel or device backend is enabled")
```

Comment:
```suggestion
    MESSAGE(FATAL_ERROR "Kokkos_ENABLE_ATOMICS_BYPASS can only be enabled if neither a host parallel nor a device backend is enabled!")
```

Reply:
From Christian "Make atomics non-atomic for non-threaded MPI-only use cases"

Reply:
Updated.  Let me know if you still don't like it.